velox/core/QueryCtx.h

```diff
@@ -200,38 +207,54 @@ class ExecCtx {
     decodedVectorPool_.push_back(std::move(vector));
   }
 
-  VectorPool& vectorPool() {
-    return vectorPool_;
+  VectorPool* vectorPool() {
+    return vectorPool_.get();
   }
 
   /// Gets a possibly recycled vector of 'type and 'size'. Allocates from
   /// 'pool_' if no pre-allocated vector.
   VectorPtr getVector(const TypePtr& type, vector_size_t size) {
-    return vectorPool_.get(type, size);
+    if (vectorPool_) {
+      return vectorPool_->get(type, size);
+    } else {
+      return BaseVector::create(type, size, pool_);
+    }
   }
 
   /// Moves 'vector' to the pool if it is reusable, else leaves it in
   /// place. Returns true if the vector was moved into the pool.
   bool releaseVector(VectorPtr& vector) {
-    return vectorPool_.release(vector);
+    if (vectorPool_) {
+      return vectorPool_->release(vector);
+    }
+    return false;
   }
 
   /// Moves elements of 'vectors' to the pool if reusable, else leaves them
   /// in place. Returns number of vectors that were moved into the pool.
   size_t releaseVectors(std::vector<VectorPtr>& vectors) {
-    return vectorPool_.release(vectors);
+    if (vectorPool_) {
+      return vectorPool_->release(vectors);
+    }
+    return 0;
+  }
+
+  bool enableOperatorBufferCache() const {
```

Comment:
```
s/enableOperatorBufferCache/enableCache/ as this is for eval ctx? 
s/enableOperatorBufferCache_/enableCache_/
thanks!
```

Reply:
> enableOperatorBufferCache

This name suggest the method enables the cache. Consider renaming to isXxxCacheEnabled.

Wondering what "buffer" means here. It feels redundant to include both "buffer" and "cache". How about isOperatorCacheEnabled?

Any particular reason we do not disable DecodedVector and SelectivityVector pools as well?

Would it make sense to rename the flag to be expression evaluation specific? 

expression_evaluation_caches_enabled

CC: @bikramSingh91

Reply:
Hi @mbasmanova, isOperatorCacheEnabled sounds good to me. 

> Any particular reason we do not disable DecodedVector and SelectivityVector pools as well?

The reason we don't disable DecodedVector and SelectivityVector pool right away is that we need to evaluate their performance impact first. I'm planning to make a draft diff on top of this PR that disables those two pools and ask the customer to run their pipelines to confirm there is no performance regression. Since there is uncertainty of disabling those two pools and they take smaller memory than the VectorPool and evalWithMemo cache disabled in this PR, I'd like to make them separate changes.

> Would it make sense to rename the flag to be expression evaluation specific?

VectorPool is not specific to expression evaluation. It's part of ExecCtx inside OperatorCtx, so every operator can have a VectorPool.